cmd/piper.go

```diff
@@ -365,8 +365,12 @@ func convertValueFromString(config map[string]interface{}, optionsField *reflect
 func convertValueFromFloat(config map[string]interface{}, optionsField *reflect.StructField, paramName string, paramValue float64) error {
 	switch optionsField.Type.Kind() {
 	case reflect.String:
-		config[paramName] = strconv.FormatFloat(paramValue, 'f', -1, 64)
-		return nil
+		val := strconv.FormatFloat(paramValue, 'f', -1, 64)
+		if strings.Contains(val, ".") {
```

Comment:
I think you should add a comment here one why you check for a period. I mean the effect should be that now also regular integers are prevented to be converted to string, right? Is this the intention here, or do you take the absence of a period as indication for an overflowing number? All I'm saying is that there should be a comment clarifying this or explaining the trade-offs.